frontend/src/lib/layout/Breadcrumbs.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+  import { page } from '$app/stores';
+  import t from '$lib/i18n';
+  import type { ProjectPageData } from '$lib/page-data';
+
+  interface Breadcrumb {
+    name: string;
+    href: string;
+  }
+  let crumbs: Breadcrumb[] = [];
+
+  type CrumbMapperFunc = (token: string) => string;
+  type CrumbValue = string | CrumbConfig | CrumbMapperFunc | undefined | null;
+  type CrumbFallbackMapper = { _get: CrumbMapperFunc };
+  type CrumbWithChild = { _name: string | null };
+  interface CrumbConfig extends Record<string, CrumbValue>, Partial<CrumbFallbackMapper>, Partial<CrumbWithChild> {}
+
+  const CRUMB_CONFIG = {
+    _name: 'user_dashboard.title',
+    admin: 'admin_dashboard.title',
+    resetPassword: 'reset_password.title',
+    project: {
+      _name: null,
+      create: 'project.create.title',
+      _get: () => {
+        const data = $page.data as ProjectPageData;
+        return data.project?.name ?? data.code;
```

Comment:
this isn't my favorite way of doing this as it's quite fragile and would be easy to forget to change this code here. It works well enough for now though.